File: CS/Basic/recursion/SwapPosition.java
package CS.Basic.recursion;

public class SwapPosition {
    public static String swapPosition(String s) {
        
        System.out.println(s.length());
        if (s.length() <= 1) {
            return s;
        } 
        else if (s.length() == 2) {
            return s.substring(1, 2) + s.substring(0, 1);
        }

        return s.substring(1, 2) + s.substring(0, 1) + swapPosition(s.substring(2, s.length()));
    }

    public static void main(String[] args) {

        System.out.println(swapPosition("abcd"));
    }
}

File: CS/Basic/recursion/SwapPosition.go
package main

// import(
// 	"fmt"
// )

func swapPosition(s string) string {
	if len(s) <= 1 {
		return s
	}

	if len(s) == 2 {
		return string(s[1]) + string(s[0])
	}

	return string(s[1]) + string(s[0]) + swapPosition(s[2:])
}

// func main() {
// 	fmt.Println(swapPosition("badce"))
// }
